MomentumDiscordBot/Services/SlashCommandService.cs

```diff
@@ -52,5 +62,65 @@ private Task _commands_SlashCommandErrored(SlashCommandsExtension sender, SlashC
 
             return Task.CompletedTask;
         }
+
+        private Task _discordClient_GuildsDownloaded(DiscordClient sender, GuildDownloadCompletedEventArgs e)
+        {
+
+            _ = Task.Run(async () =>
+            {
+                _textChannel = await _discordClient.GetChannelAsync(_config.AdminBotChannel);
+                await FindRestartMessageAsync();
+            });
+
+            return Task.CompletedTask;
+        }
+
+        private async Task FindRestartMessageAsync()
+        {
+            // Filter only messages from this bot
+            var existingMessages = (await _textChannel.GetMessagesAsync()).FromSelf(_discordClient);
```

Comment:
It would be good to explicitly state the `limit` value passed in `GetMessagesAsync`, I was worried this was going to try and read every message the bot has ever posted. Also we could probably safely have the limit lower than 100, maybe 50 is better.